index.js

```diff
@@ -200,8 +203,18 @@ function adapter(uri, opts){
 
   Redis.prototype.del = function(id, room, fn){
     debug('removing %s from %s', id, room);
+    var hasRoom = this.rooms.hasOwnProperty(room);
     Adapter.prototype.del.call(this, id, room);
-    runCommand('unsubscribe', this.getChannelName(room));
+    if (hasRoom && !this.rooms[room]) {
```

Comment:
Isn't this duplicative of Adapter's del() function?

https://github.com/socketio/socket.io-adapter/blob/master/index.js#L61-L70